File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/asnytasks/SearchManAsyncTask.java
package com.inspiredandroid.linuxcommandbibliotheca.asnytasks;

import android.content.Context;
import android.os.AsyncTask;

import com.inspiredandroid.linuxcommandbibliotheca.interfaces.OnConvertFromHtmlToSpannableListener;
import com.inspiredandroid.linuxcommandbibliotheca.misc.Utils;

import java.util.ArrayList;

/**
 * Created by Simon Schubert
 */
public class SearchManAsyncTask extends AsyncTask<String, ArrayList<ArrayList<CharSequence>>, ArrayList<ArrayList<CharSequence>>> {

    private OnConvertFromHtmlToSpannableListener mCallback;
    private Context mContext;
    private ArrayList<ArrayList<CharSequence>> mChildren;
    private String mQuery;

    public SearchManAsyncTask(Context context, String query, ArrayList<ArrayList<CharSequence>> children, OnConvertFromHtmlToSpannableListener callback) {
        mContext = context;
        mChildren = children;
        mCallback = callback;
        mQuery = query;
    }

    @Override
    protected ArrayList<ArrayList<CharSequence>> doInBackground(String... params) {
        for (ArrayList<CharSequence> children : mChildren) {
            for (int i = 0; i < children.size(); i++) {
                CharSequence chld = children.get(i);
                children.set(i, Utils.highlightQueryInsideText(mContext, mQuery, chld.toString()));
            }
        }

        return mChildren;
    }

    @Override
    protected void onPostExecute(ArrayList<ArrayList<CharSequence>> children) {
        super.onPostExecute(children);

        mCallback.onConvertHtmlToSpannable(children);
    }

}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/asnytasks/SearchManAsyncTask.kt
package com.inspiredandroid.linuxcommandbibliotheca.asnytasks

import android.content.Context
import android.os.AsyncTask
import com.inspiredandroid.linuxcommandbibliotheca.interfaces.OnConvertFromHtmlToSpannableListener
import com.inspiredandroid.linuxcommandbibliotheca.misc.Utils
import java.util.*

/**
 * Created by Simon Schubert
 */
class SearchManAsyncTask(private val mContext: Context, private val mQuery: String, private val mChildren: ArrayList<ArrayList<CharSequence>>, private val mCallback: OnConvertFromHtmlToSpannableListener) : AsyncTask<String, ArrayList<ArrayList<CharSequence>>, ArrayList<ArrayList<CharSequence>>>() {

    override fun doInBackground(vararg params: String): ArrayList<ArrayList<CharSequence>> {
        for (children in mChildren) {
            for (i in children.indices) {
                val chld = children[i]
                children[i] = Utils.highlightQueryInsideText(mContext, mQuery, chld.toString())
            }
        }

        return mChildren
    }

    override fun onPostExecute(children: ArrayList<ArrayList<CharSequence>>) {
        super.onPostExecute(children)

        mCallback.onConvertHtmlToSpannable(children)
    }

}


